extensions-core/lookups-cached-global/src/main/java/org/apache/druid/server/lookup/namespace/cache/CacheScheduler.java

```diff
@@ -197,6 +197,17 @@ public void run()
       }
     }
 
+    public VersionedCache createFromExistingCache(
+        @Nullable EntryImpl<? extends ExtractionNamespace> entryId,
+        String version,
+        Map<String, String> newCacheEntries
+    )
+    {
+      VersionedCache state = (VersionedCache) cacheStateHolder.get();
```

Comment:
A closing race (from `NamespaceLookupExtractorFactory.close()`) may lead to `cacheStateHolder` holding `NoCache`. Please handle it gracefully by returning `@Nullable` from this method and the calling `generateCache()` method. Add a Javadoc comment explaining the race condition.

Reply:
added javadoc comment and method now returns @Nullable. `generateCache()` was already returning @Nullable.

Reply:
How about only update the original VersionedCache's version and  return it back, when the previous CacheState is not instance of NoCache. So we needn't care about closing race.
Then check if the generated cache is same as cacheStateHolder, treat updatedCacheSuccessfully as true.